### ISSUE TITLE
Tests fail in Node 4.0.0

### ISSUE THREAD

**Issue by emgee3** [laverdet/node-fibers#242]
##### tl;dr: `stackoverflow-2` and `stackoverflow-3` tests fail in Node 4.0.0 on both OS X and Ubuntu.

Fresh install of Ubuntu Server 14.04, running Node 4.0.0, which uses a new version of V8.

It does compile and install correctly: 

```
root@nodetest:~/t# npm install --save fibers
npm WARN package.json t@1.0.0 No description
npm WARN package.json t@1.0.0 No repository field.
npm WARN package.json t@1.0.0 No README data
-
> fibers@1.0.6 install /root/t/node_modules/fibers
> node build.js || nodejs build.js

make: Entering directory `/root/t/node_modules/fibers/build'
  CXX(target) Release/obj.target/fibers/src/fibers.o
  CXX(target) Release/obj.target/fibers/src/coroutine.o
  CC(target) Release/obj.target/fibers/src/libcoro/coro.o
  SOLINK_MODULE(target) Release/obj.target/fibers.node
  COPY Release/fibers.node
make: Leaving directory `/root/t/node_modules/fibers/build'
Installed in `/root/t/node_modules/fibers/bin/linux-x64-v8-4.5/fibers.node`
fibers@1.0.6 node_modules/fibers
```

However running `npm test` shows failed tests:

```
root@nodetest:~/t/node_modules/fibers# npm test

> fibers@1.0.6 test /root/t/node_modules/fibers
> node test.js || nodejs test.js

already-running.js: pass
bad-context.js: pass
child-process.js: pass
current.js: pass
exec.js: pass
exit.js: pass
fibonacci.js: pass
finish-multiple.js: pass
future-exception.js: pass
future.js: pass
illegal-yield.js: pass
pool.js: pass
process-title.js: pass
stack-overflow.js: pass
stack-overflow2.js: *fail*
code: null
stderr: 
stdout: 
stack-overflow3.js: *fail*
code: null
stderr: 
stdout: 
started.js: pass
unwind.js: pass
already-running.js: pass
bad-context.js: pass
child-process.js: pass
current.js: pass
exec.js: pass
exit.js: pass
fibonacci.js: pass
finish-multiple.js: pass
future-exception.js: pass
future.js: pass
illegal-yield.js: pass
pool.js: pass
process-title.js: pass
stack-overflow.js: pass
stack-overflow2.js: *fail*
code: null
stderr: 
stdout: 
stack-overflow3.js: *fail*
code: null
stderr: 
stdout: 
started.js: pass
unwind.js: pass
npm ERR! Test failed.  See above for more details.
root@nodetest:~/t/node_modules/fibers# 
```

The same tests fail in OS X El Capitan.

**Comment by Gaubee** [comment]
Windows is OK.
Ubuntu GG~

**Comment by laverdet** [comment]
The stack-overflow tests are known to be kind of finicky and don't really represent real-world usage of fibers so you *should* be fine be running your application on node v4.x. I've looked into this a lot and v8's behavior when approaching stack boundaries is pretty inconsistent erratic. That said I'll definitely need to take a look at this.

**Comment by seeekr** [comment]
May I hijack this thread a bit and ask what the current status is for node-fibers and the latest versions of Node? (4.x) I understand that I could just build it and see for myself, but what I'm interested in is the project owner's / collaborators' view. Is it already fully supported and only lacking battle-testing from real-world use? And otherwise actively maintained as one would generally expect?

(Background: I'm interested in seeing Meteor move over to the new versions of Node and that's highly dependent on node-fibers and a couple other dependencies.)

**Comment by laverdet** [comment]
fibers is pretty well battle tested in general. Upgrades to node don't really matter, it's v8 you have to worry about. If it compiles for you, there's no reason it shouldn't be just as stable as it was in previous versions of v8. Obviously I wouldn't switch production code over to it with no testing, but I don't foresee you running into any issues.

**Comment by laverdet** [comment]
These tests were fixed in ea63217cbe1806d2d209907ccbca4d0e64a0016d and 3e571a7fe9c8713f95c3e9c0305af3c8208b24b2.
